index.js

```diff
@@ -1,383 +1,81 @@
 const PMIDSset = new Set();
 let instructions = [];
 let dropIndexes = false;
+let prefix = '' ;
+let outputPath = '';
+let outDir = './data';
+
 
 const parseModelFiles = (modelDir) => {
   // find the yaml in the folder
-  yamlFile = getFile(modelDir, /.*[.](yaml|yml)$/);
+  yamlFile = utils.getFile(modelDir, /.*[.](yaml|yml)$/);
   if (!yamlFile) {
     throw new Error("yaml file not found in path ", modelDir);
   }
 
-  const [ metadata, metabolites, reactions, genes, compartments ] = yaml.safeLoad(fs.readFileSync(yamlFile, 'utf8'));
-  const metadataSection = metadata.metaData || metadata.metadata;
-  const model = toLabelCase(metadataSection.short_name);
-  const version = `V${metadataSection.version.replace(/\./g, '_')}`;
-  const isHuman = metadataSection.short_name === 'Human-GEM';
+  const [metadata, metabolites, reactions, genes, compartments, metadataSection, model, version, isHuman] = parser.getInfoFromYaml(yamlFile);
 
-  const prefix = `${model}${version}`;
-  const outputPath = `./data/${prefix}.`;
+  prefix = `${model}${version}`;
+  outputPath = `${outDir}/${prefix}.`;
 
   content = { // reformat object as proper key:value objects, rename/add/remove keys
-    compartmentalizedMetabolite: reformatCompartmentalizedMetaboliteObjets(metabolites.metabolites),
-    reaction: reformatReactionObjets(reactions.reactions),
-    gene: reformatGeneObjets(genes.genes),
-    compartment: reformatCompartmentObjets(compartments.compartments),
-  }
-
-  const componentIdDict = {}; // store for each type of component the key  Id <-> element
-  // use to filter out annotation/external ids for components not in the model and to add missing information
-  // extracted from these annotation files such as description, etc...
-  Object.keys(content).forEach((k) => {
-    componentIdDict[k] = Object.fromEntries(content[k].map(e => [e[`${k}Id`], e]));
-  });
-
-  if (isHuman) {
-    Object.keys(componentIdDict.gene).forEach((geneId) => {
-      humanGeneIdSet.add(geneId);
-    });
+    compartmentalizedMetabolite: utils.reformatCompartmentalizedMetaboliteObjets(metabolites.metabolites),
+    reaction: utils.reformatReactionObjets(reactions.reactions),
+    gene: utils.reformatGeneObjets(genes.genes),
+    compartment: utils.reformatCompartmentObjets(compartments.compartments),
   }
 
-  // subsystems are not a section in the yaml file, but are extracted from the reactions info
-  content.subsystem = [];
-  componentIdDict.subsystem = {};
-  content.reaction.forEach((r) => {
-    r.subsystems.forEach((name) => {
-      const id = idfyString(name);
-      const subsystemObject = { subsystemId: id, name }; // TODO add 'description' key
-      if (!(id in componentIdDict.subsystem)) {
-        content.subsystem.push(subsystemObject);
-        componentIdDict.subsystem[id] = subsystemObject;
-      };
-    });
-  });
+  const componentIdDict = utils.getComponentIdDict(content, isHuman);
+  utils.getHumanGeneIdSet(componentIdDict, isHuman, humanGeneIdSet);
```

Comment:
Sorry that I missed this in the previous review. I'm not sure if the `componentIdDict` gets updated with the refactoring of the `getHumanGeneIdSet` function.

Reply:
I think the value of `componentIdDict` will not be changed by moving getHumanGeneIdSet out of the function. But I can double check.

Reply:
I can confirm that the value of `componentIdDict` is the same after refactoring.

Reply:
Thanks for confirming. Just to double check, this means we should update the `index.js` as well as the `utils.js` right?

Reply:
Yes. Both `index.js` and `utils.js` have been updated by the commit c6ec8d9

Reply:
I took another look at this and I think I misunderstood something. To me it looked like when the `getHumanGeneIdSet` function in `utils.js` gets called it would not do anything. But after testing I see that it actually populates the `humanGeneIdSet` variable inside `index.js`. I'm used to having functions not modify variables outside of it so I didn't think that the `humanGeneIdSet` would actually get modified globally. So I think all the data output is correct regarding to this.

If you would like to, I suggest the following refactoring changes. They are not directly related to my initial concern though so I think I could resolve this conversation and approve the PR.

```javascript
  const componentIdDict = utils.getComponentIdDict(content);
  if (isHuman) {
    utils.getHumanGeneIdSet(componentIdDict, isHuman, humanGeneIdSet);
  }
```

```javascript
const getComponentIdDict = (content) => {
 ...
}

const getHumanGeneIdSet = (componentIdDict, humanGeneIdSet) => {
  // the parameter humanGeneIdSet will be updated
  Object.keys(componentIdDict.gene).forEach((geneId) => {
    humanGeneIdSet.add(geneId);
  });
}
```

Reply:
Great that you spotted it @e0, `isHuman` is not needed by `getComponentIdDict` anymore since the code to getHumanGeneIdSet has been moved out, and I think it is more logical to check `isHuman` in `index.js` before calling `getHumanGeneIdSet()`. This refactoring has been addressed in the commit 7b37d1e

Reply:
That's great, thank you for updating it!